src/context/delay.ts

```diff
@@ -16,16 +17,54 @@ const getRandomServerResponseTime = () => {
   )
 }
 
+type DelayMode = 'real' | 'infinite'
+
 /**
  * Delays the response by the given duration (ms).
  * @example
- * res(ctx.delay()) // realistic server response time
- * res(ctx.delay(1200))
+ * res(ctx.delay(1200)) // delay response by 1200ms
+ * res(ctx.delay()) // emilate realistic server response time
+ * res(ctx.delay('infinite')) // delay response infinitely
  * @see {@link https://mswjs.io/docs/api/context/delay `ctx.delay()`}
  */
-export const delay = (durationMs?: number): ResponseTransformer => {
+export const delay = (
+  durationOrMode?: DelayMode | number,
+): ResponseTransformer => {
   return (res) => {
-    res.delay = durationMs ?? getRandomServerResponseTime()
+    let delayTime: number
+
+    if (typeof durationOrMode === 'string') {
+      switch (durationOrMode) {
+        case 'infinite': {
+          // Using `Infinity` as a delay value executes the response timeout immediately.
+          // Instead, use the maximum allowed integer for `setTimeout`.
+          delayTime = SET_TIMEOUT_MAX_ALLOWED_INT
```

Comment:
I think these `case` statements need a `break` in between, otherwise it cascades and end up by throwing an error.